odoo/addons/base/models/ir_qweb.py

```diff
@@ -610,19 +618,169 @@ def _render(self, template, values=None, **options):
+                    raise RecursionError('Qweb template infinity loop')  # noqa: TRY301
+
+                dont_fetch_directive = frame.irQweb.env.context.get('__qweb_dont_fetch_directive')
+
+                # traverse the iterator
+                for item in frame.iterator:
+                    if isinstance(item, str) or item.directive == dont_fetch_directive:
+                        yield item
+                    else:
+                        # add new QwebStackFrame from QwebCallParameters
+                        values = frame.values
+                        irQweb = frame.irQweb
+                        params = item
+
+                        render_template = loaded_functions.get(params.method)
+
+                        # Use the current directive context
+                        if params.context:
+                            irQweb = irQweb.with_context(**params.context)
+
+                        # Fetch the compiled function and template options
+                        if not render_template:
+                            template_functions, def_name, options = irQweb._compile(params.view_ref)
+                            loaded_functions.update(template_functions)
+                            render_template = template_functions[params.method or def_name]
+                        else:
+                            options = irQweb._compile(params.view_ref)[2]
+
+                        # Apply a new scope if needed
+                        if params.scope:
+                            if params.scope == 'root':
+                                values = root_values
+                            values = values.copy()
+
+                        # Update values ​​with default values
+                        if params.values:
+                            values.update(params.values)
+
+                        iterator = None
+                        try:
+                            # Create the iterator from the template
+                            iterator = render_template(irQweb, values)
+                        finally:
+                            stack.append(QwebStackFrame(params, irQweb, iterator, values, options))
+                        break
+                else:
+                    stack.pop()
+
+            except (TransactionRollbackError, ReadOnlySqlTransaction):
+                raise
+            except Exception as error:
+                qweb_error_info = self._get_error_info(error, stack, frame)
+
+                if hasattr(error, 'qweb'):
+                    if qweb_error_info.source:
+                        error.qweb.source = f"{error.qweb.source}\n      {qweb_error_info.source}"
+                    if not error.qweb.ref and frame.params.view_ref:
+                        error.qweb.ref = frame.params.view_ref
+                    qweb_error_info = error.qweb
+                elif not isinstance(error, tuple(Exception for name, Exception in exceptions.__dict__.items() if not name.startswith('_'))):
```

Comment:
I think you just need to check `UserError` (other errors are subtypes or seem internal).